### PR TITLE
[Merged by Bors] - chore(Algebra/CharP/Basic): Split

### PR DESCRIPTION
**Pull request by YaelDillies**
`Algebra.CharP.Basic` currently depends on 935 files. This is unacceptably large for such a basic plain algebra file. This PR moves everything that doesn't depend on `Finset` (a seemingly good cut-off point) to a new file `Algebra.CharP.Defs`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

**Comment by YaelDillies** [comment]
Probably worth waiting for the shake before benching!

**Comment by mattrobball** [comment]
Oh I see. With `awaiting-review`, I assumed that this had been done and shake would clean up the stragglers. 

Anyway, we can bench it again afterwards.

File: Mathlib/Algebra/CharP/Defs.lean [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+  | inr h => have : CharP R 0 := h ▸ inferInstance; exact CharP.charP_to_charZero R
+
+end CharZero
+
+namespace Fin
+
+instance charP (n : ℕ) : CharP (Fin (n + 1)) (n + 1) where
+    cast_eq_zero_iff' := by simp [Fin.ext_iff, Nat.dvd_iff_mod_eq_zero]
+
+end Fin
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/CharP/Defs.lean:457]: I'd consider keeping this in Basic

Comment (YaelDillies) [reply Mathlib/Algebra/CharP/Defs.lean:457]: Why? `Fin` is in Std now

**Comment by YaelDillies** [comment]
Sorry, I use `awaiting-CI` + `awaiting-review` to mean "`awaiting-review` once `awaiting-CI` is removed". That spares me one manual label manipulation.

**Comment by mattrobball** [comment]
!bench

File: Mathlib/Algebra/CharP/Defs.lean (hunk 1 above)
Comment (YaelDillies) [reply Mathlib/Algebra/CharP/Defs.lean:457]: @Ruben-VandeVelde, in my future PRs I need the instance above and I don't need the junk from `Algebra.CharP.Basic` (which still needs sorting). I think that's a good enough reason to move `Fin.charP` to the more basic file. Further, it does not increase imports.

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

P.S.: If someone wants to split the file into smaller pieces, this can be done in another PR.

**Comment by Vierkantor** [comment]
This split is a good idea! Unfortunately the `Defs` file in this PR doesn't work for my use case (#10765, which has been rotting without comments for months), since it still imports the `ring` tactic. I suppose I'll make a new PR that also splits out everything depending on `import Mathlib.Data.Int.ModEq` which transitively depends on `ring`.

**Comment by Vierkantor** [comment]
(I am slightly annoyed that I didn't have the chance to complain about this before it got merged, but that's my own life getting in the way these last weeks.)

**Comment by YaelDillies** [comment]
Oh sorry I didn't know about your PR!

**Pull request closed without merging**